docs/source/opkg-keyrings/opkg-keyrings_index.rst

```diff
@@ -0,0 +1,153 @@
+    * These keys are placed in specific directories, such as /usr/share/opkg/keyrings/.
+
+#. Ensure Package Integrity:
+
+    * During installation, opkg checks package signatures using these keys.
+    * This prevents unauthorized or tampered packages from being installed.
+
+#. Key Management:
+
+    * The opkg-keyrings package is updated when signing keys are rotated (e.g., when
+        old keys are retired, and new ones are issued).
+    * Upgrading this package ensures the keyring remains current and secure.
+
+How opkg signing works and why we do it
+=======================================
+This involves cryptographic signatures that
+verify packages come from a trusted source and have not been tampered with. 
+
+How does it work?
+-----------------
+The package feeds are signed with a private key and the corresponding
+public key is included in the opkg-keyrings package. When the feed is
+updated, opkg checks the feed signature against the public key to
+verify its authenticity and integrity.
+
+Steps involved:
+----------------
+
+#. Generate and Sign the Package Index:
+
+    * A pair of private and public keys is created using a tool like gpg.
+    * The private key is used to sign the package index (package index, 
+      the *Packages* and *Packages.gz* files in the case of NILRT).
+    * A detached signature (*Packages.asc*) is generated and placed alongside the
+        *Packages* file.
+    * The public key is distributed to the target systems.
+
+#. Verify Signature When a Feed is Updated:
+
+    * When opkg updates a package feed, it verifies the signature of the *Packages* file
+      using the public key.
+    * The public key is preloaded or added to the system's trusted keyring, such as
+      opkg-key (see :ref:`How keys are added to the opkg-keyring` for more details
+      on how this is accomplished).
+
+#. Trust Management:
+
+    * Trusted keys are stored in a keyring directory (e.g., /usr/share/opkg/keyrings/).
+    * While updating the feed (running `opkg udpate`), opkg verifies the signature.
+
+.. _How keys are added to the opkg-keyring:
+
+How keys are added to the opkg-keyring
+======================================
+The Base System Image includes the signing keys, typically located at /usr/share/opkg/keyrings,
+preloaded into the keyring.
+
+However, if the signing keys used by the feeds are rotated, the target system requires an update
+to add the new key and remove any obsolete keys no longer in use.
+
+The latest version of the opkg-keyrings package, available on the distribution feed
+(https://download.ni.com/#ni-linux-rt/feeds/dist/), is built with the updated signing keys.
+Upgrading this package on the target system installs the new signing key to
+/usr/share/opkg/keyrings, updates the keyring by adding the new key, and
+removes any deprecated keys no longer used to sign the feeds.
+
+By upgrading the opkg-keyrings package, the system ensures its keyring remains up-to-date
```

Comment:
> the system ensures its keyring remains

> seamlessly handling the addition

Seeing that we're asking the user to do upgrade the package manually, mentioning "seamless" is probably not accurate. I'd prefer if we just removed the whole line 152, 153.

Reply:
I have removed this now.